src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs

```diff
@@ -395,10 +402,12 @@ public CosmosDBAccountImpl WithIpRangeFilter(string ipRangeFilter)
             return this;
         }
 
-        public CosmosDBAccountImpl WithWriteReplication(Region region)
+        public CosmosDBAccountImpl WithWriteReplication(Region region, bool? isZoneRedundant)
         {
-            Models.FailoverPolicy FailoverPolicy = new Models.FailoverPolicy();
+            Models.Location FailoverPolicy = new Models.Location();
             FailoverPolicy.LocationName = region.Name;
+            FailoverPolicy.IsZoneRedundant = isZoneRedundant;
+            FailoverPolicy.FailoverPriority = this.failoverPolicies.Count;
             this.hasFailoverPolicyChanges = true;
```

Comment:
Not sure about the purpose of `hasFailoverPolicyChanges`. Is it kind of `isFailoverPolicyLoaded`?

Reply:
Yeah, I think it is the check about loaded. Is it better to change these naming? Like the failoverPolicy actually is changed to use the Location type rather than FailoverPolicy

Reply:
This one better get changed if you got chance, I thought that it is a flag about whether one need to send a Update request if this changes, then finding out that it about loaded.

`failoverPolicy` is kind of OK, it got a class so people could always check the class.